src/app/AttributePathExpandIterator.cpp

```diff
@@ -101,15 +106,17 @@ void AttributePathExpandIterator::PrepareAttributeIndexRange(const ClusterInfo &
 {
     if (aClusterInfo.HasWildcardAttributeId())
     {
-        mAttributeIndex    = 0;
-        mEndAttributeIndex = emberAfGetServerAttributeCount(aEndpointId, aClusterId);
+        mAttributeIndex       = 0;
+        mEndAttributeIndex    = emberAfGetServerAttributeCount(aEndpointId, aClusterId);
+        mGlobalAttributeIndex = 0;
     }
     else
     {
         mAttributeIndex = emberAfGetServerAttributeIndexByAttributeId(aEndpointId, aClusterId, aClusterInfo.mAttributeId);
         // If the given attribute id does not exist on the given endpoint, it will return uint16(0xFFFF), then endAttributeIndex
         // will be 0, means we should iterate a null attribute set (skip it).
-        mEndAttributeIndex = static_cast<uint16_t>(mAttributeIndex + 1);
+        mEndAttributeIndex    = static_cast<uint16_t>(mAttributeIndex + 1);
+        mGlobalAttributeIndex = ArraySize(GlobalAttributesNotInMetadata);
```

Comment:
What if the request path is a global attribute?

IIRC reading Wildcard cluster id with global attribute id is legal.

Reply:
Good catch!  Let me sort that out.